tasks/collect-slack-notification-params/collect-slack-notification-params.yaml

```diff
@@ -0,0 +1,185 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: collect-slack-notification-params
+  labels:
+    app.kubernetes.io/version: "0.0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task that collects slack notification configuration options from Release CRs and the data file
+  params:
+    - name: releasePath
+      type: string
+      description:
+        Path to the JSON string of the Release CR
+    - name: snapshotPath
+      description: Path to the JSON string of the mapped Snapshot spec in the data workspace
+      type: string
+      default: snapshot_spec.json
+    - name: dataPath
+      type: string
+      description: Path to the JSON string of the merged data containing the configuration options to use
+      default: data.json
+    - name: hacUrl
+      type: string
+      description: Root URL of HAC UI (hybrid access console)
+      default: "https://console.redhat.com/preview/application-pipeline/workspaces"
+  workspaces:
+    - name: data
+      description: The workspace where the snapshot spec json file resides
+  results:
+    - name: message
+      type: string
+      description: the formatted message to use in downstream tasks
+    - name: slack-notification-secret
+      type: string
+      description: Name of secret which contains keyname=webhookURLs
+    - name: slack-notification-secret-keyname
+      type: string
+      description: Name of key within secret which contains webhook URL
+  steps:
+    - name: create-message
+      image: quay.io/hacbs-release/release-utils:85ab98a7ec63c3d8d9ec3c4982ff0e581bcb3c83
+      script: |
+        #!/usr/bin/env sh
+        set -x
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+        SNAPSHOT_FILE="$(workspaces.data.path)/$(params.snapshotPath)"
+        if [ ! -f "${SNAPSHOT_FILE}" ] ; then
+            echo "No valid snapshot file was provided."
+            exit 1
+        fi
+        RELEASE_FILE="$(workspaces.data.path)/$(params.releasePath)"
+        if [ ! -f "${RELEASE_FILE}" ] ; then
+            echo "No valid release file was provided."
+            exit 1
+        fi
+
+        if [ $(jq '.slack | has("slack-notification-secret")' $DATA_FILE) == false ] ; then
+            echo "No secret name provided via 'slack.slack-notification-secret' key in Data."
+            echo -n "" > $(results.message.path)
+            echo -n "" > $(results.slack-notification-secret.path)
+            echo -n "" > $(results.slack-notification-secret-keyname.path)
+            exit
+        fi
+        slack_notification_secret=$(jq -r '.slack."slack-notification-secret"' $DATA_FILE)
```

Comment:
I was gonna complain about the `""` - it seems weird. But wow, it's actually needed. I learnt something new today :)

Apparently without it, jq will treat `-` as a subtraction.